src/fil_cns/validation.rs

```diff
@@ -197,10 +195,18 @@ fn block_timestamp_checks(
     base_tipset: &Tipset,
     chain_config: &ChainConfig,
 ) -> Result<(), FilecoinConsensusError> {
+    if header.epoch <= base_tipset.epoch() {
```

Comment:
Add this check to be on par with lotus. See https://github.com/filecoin-project/lotus/blob/e68b762b40cd650410d5b203c578e196f32427d9/chain/consensus/filcns/filecoin.go#L139

```go
// fast checks first
	if h.Height <= baseTs.Height() {
		return xerrors.Errorf("block height not greater than parent height: %d != %d", h.Height, baseTs.Height())
	}
```